doc/langdef.md

```diff
@@ -441,6 +437,38 @@ Also note that all of these conversions are dynamic at runtime, so CEL's static
 type analysis cannot avoid the possibility of type-related errors in expressions
 using these dynamic values.
 
+## JSON Data Conversion
+
+CEL can also work with JSON data. Since there is a natural correspondence of
+most CEL data with protocol buffer data, and protocol buffers have a
+[defined mapping](https://developers.google.com/protocol-buffers/docs/proto3#json)
+to JSON, this creates a natural mapping of CEL to JSON. This creates an exact
+bidirectional mapping between JSON types and a subset of CEL data:
+
+JSON Type | CEL Type
+--------- | ----------------------------------------------
+`null`    | `null`
+Boolean   | `bool`
+Number    | `double` (except infinities or NaN)
+String    | `string`
+Array     | `list` of elements in this table
+Object    | `map` (with string keys, values in this table)
+
+We define JSON mappings for much of the remainder of CEL data, but note that
+this data will not map back in to CEL as the same value:
+
+CEL Data                                             | JSON Data
+---------------------------------------------------- | ---------
+`int`                                                | Number if in interoperable range, otherwise decimal String.
```

Comment:
Let's mention that the interoperable range is -(2^53 - 1) to 2^53 - 1.

Reply:
Done.